scripts/generate-patch

```diff
@@ -27,7 +27,7 @@ for f in packages/*; do
 				curl -sLf ${url} | tar xvzf - -C ${f}/charts-original --strip ${fields} ${subdirectory} > /dev/null 2>&1
 			fi
 			if [[ -d ${f}/charts ]]; then
-				diff -uNr ${f}/charts-original ${f}/charts > ${f}/$(basename -- ${f}).patch || true
+				diff -x *.tgz -uNr ${f}/charts-original ${f}/charts > ${f}/$(basename -- ${f}).patch || true
```

Comment:
Is this still needed? What would be the case where tgz file exists on the chart after running `helm dependency update`?

Reply:
It depends on how the chart itself stores the dependencies. 

Case 1: For a chart like [rancher-monitoring (v1)](https://github.com/rancher/system-charts/tree/dev-v2.5/charts/rancher-monitoring/v0.1.1) that stores the chart [as files](https://github.com/rancher/system-charts/blob/dev-v2.5/charts/rancher-monitoring/v0.1.1/requirements.yaml), the prepare script will populate the charts/ directory with the subchart directories themselves.

Case 2: For a chart like prometheus-operator that declares dependencies on charts [in other repositories](https://github.com/helm/charts/blob/master/stable/prometheus-operator/requirements.yaml), the default is to retrieve them as chart archives (i.e. .tgz files). This is noted in [the Helm docs](https://helm.sh/docs/topics/charts/): "when `helm dependency update` retrieves charts, it will store them as chart archives in the charts/ directory."

Reply:
For Case 1, I think it makes sense to allow dev-charts to define patches on the subcharts since the subcharts are defined directly in the parent charts. However, for Case 2, it doesn't make sense to define a patch on a binary so I think it should be safe to omit the .tgz files; we can still track if there are changes to the dependencies in the patch based on the changes to `requirements.lock` and `requirements.yaml` seen [here](https://github.com/aiyengar2/dev-charts/blob/4ecef3ea9f879dc4d416618a2f3666cdffd95538/packages/rancher-monitoring-v2/rancher-monitoring-v2.patch#L17-L47). If patches are required to these types of subcharts, they should ideally be in a different dev-chart and the parent chart's dependencies should be updated to point to the dev-chart version of the subchart.